src/main/java/org/folio/services/finance/transaction/BaseTransactionService.java

```diff
@@ -90,20 +90,26 @@ public Future<Void> updateTransaction(Transaction transaction, RequestContext re
   }
 
   public Future<Void> updateTransactions(List<Transaction> transactions, RequestContext requestContext) {
-    var semaphore = new Semaphore(5, requestContext.getContext().owner());
-    List<Future<Void>> futures = new ArrayList<>();
-
+    if (CollectionUtils.isEmpty(transactions)) {
+      return Future.succeededFuture();
+    }
     return requestContext.getContext()
-      .executeBlocking(promise -> {
+      .<List<Future<Void>>>executeBlocking(promise -> {
+        List<Future<Void>> futures = new ArrayList<>();
+        var semaphore = new Semaphore(1, requestContext.getContext().owner());
```

Comment:
Also I had some concerns that if we have for example 5 concurrent users - the processing of request for them will be sequnetial, but we checked that object request.getContext().owner() is different from request to request, it means for concurrent users new semaphores will be created locked on different objects, so for different users it will be parallel processing as we desired, for single user -> all transactions will be processed sequnetially